core/controllers/acl_decorators.py

```diff
@@ -2174,7 +2174,10 @@ def test_can_view_any_topic_editor(self, topic_id, **kwargs):
         """
         if not self.user_id:
             raise self.NotLoggedInException
-        topic_domain.Topic.require_valid_topic_id(topic_id)
+        try:
+            topic_domain.Topic.require_valid_topic_id(topic_id)
+        except Exception as e:
```

Comment:
@kevintab95 Should we only catch the `ValidationError` exception?

Reply:
Also, there are lots of other places in the handler where we are using `require_valid_topic_id`, should we change this method to `is_valid_topic_id` which will return boolean value?

![image](https://user-images.githubusercontent.com/16653571/75461152-598cd100-59a8-11ea-90f1-dffb5a348f3e.png)

Reply:
![image](https://user-images.githubusercontent.com/16653571/75461941-7aa1f180-59a9-11ea-9f54-c890adbb919e.png)
I'm a bit confused with the use-case of this "require_valid_topic_id" method.

In the above image, we are fetching the topic on line 342 and validating the topic id on line 346.

@aks681, Can you please explain why the "require_valid_topic_id" method was implemented in the first place? Do we have any similar checks for exploration/collection handlers? If not then I would suggest removing this check maybe?

Reply:
Yeah, in the screenshot you shared, it is a mistake, it should be the other way around. Ideally, all instances of this should be in acl_decorators inside the correct decorator, so @kevintab95 can you move the occurences in topic_editor to its corresponding decorator? Since, that's where we check the validity of the topic. You can raise the error there.
@DubeySandeep A separate function was written, since it was used in multiple decorators. It doesn't do much, just checks if it is a string of size 12.

Reply:
Done.